### ISSUE TITLE
Error on https://www.alphabounce.com/

### ISSUE THREAD

**Issue by 9r8k** [ruffle-rs/ruffle#2388]
Hello Ruffle team. At the moment I'm playing the game AlphaBounce under the given domain. In order to run the game, a Twinoid account has to be created (can be done quickly) such that the progress can be saved. Usually I use Flash Player with the Firefox browser on my PC. This always worked and it still works.

Although it is sometimes necessary to refresh the page a few times because sometimes the Flash application crashes on my PC. But if it does not crash in the first few seconds, it continues to work until I close the Firefox browser again.

However from tomorrow it will be blocked completely of course. So I tried Ruffle (on the same PC with the same browser) but unfortunately I get the following error message.

# Error Info
Error name: RuntimeError
Error message: unreachable executed
Error stack:
```
std::panicking::rust_panic_with_hook::h123718ba3bf480af@moz-extension://fe553814-45df-434f-9b5e-b79a2f12d8e1/dist/aababf0a1e5ed63fc484.module.wasm:wasm-function[3304]:0x33c6db
std::panicking::begin_panic_handler::{{closure}}::hf393a82e58397bd2@moz-extension://fe553814-45df-434f-9b5e-b79a2f12d8e1/dist/aababf0a1e5ed63fc484.module.wasm:wasm-function[3885]:0x357ce6
std::sys_common::backtrace::__rust_end_short_backtrace::h2ff2cfc953878925@moz-extension://fe553814-45df-434f-9b5e-b79a2f12d8e1/dist/aababf0a1e5ed63fc484.module.wasm:wasm-function[4618]:0x36733c
rust_begin_unwind@moz-extension://fe553814-45df-434f-9b5e-b79a2f12d8e1/dist/aababf0a1e5ed63fc484.module.wasm:wasm-function[4492]:0x3659ab
core::panicking::panic_fmt::h285dc015a87be5a4@moz-extension://fe553814-45df-434f-9b5e-b79a2f12d8e1/dist/aababf0a1e5ed63fc484.module.wasm:wasm-function[4620]:0x3673a0
core::option::expect_none_failed::hb95d2b1dc5448c17@moz-extension://fe553814-45df-434f-9b5e-b79a2f12d8e1/dist/aababf0a1e5ed63fc484.module.wasm:wasm-function[3961]:0x35a2f8
instant::wasm::now::he7a47d56b3f3cd75@moz-extension://fe553814-45df-434f-9b5e-b79a2f12d8e1/dist/aababf0a1e5ed63fc484.module.wasm:wasm-function[3678]:0x34fc87
ruffle_core::player::Player::new::h6ee48c139d9e22fc@moz-extension://fe553814-45df-434f-9b5e-b79a2f12d8e1/dist/aababf0a1e5ed63fc484.module.wasm:wasm-function[345]:0xcc062
ruffle_web::Ruffle::new::h8e2822ae9db87044@moz-extension://fe553814-45df-434f-9b5e-b79a2f12d8e1/dist/aababf0a1e5ed63fc484.module.wasm:wasm-function[421]:0x118fa3
ruffle_new@moz-extension://fe553814-45df-434f-9b5e-b79a2f12d8e1/dist/aababf0a1e5ed63fc484.module.wasm:wasm-function[3891]:0x357fc7
H@moz-extension://fe553814-45df-434f-9b5e-b79a2f12d8e1/dist/core.ruffle.js:1:19898
ensureFreshInstance/<@https://www.alphabounce.com/:3:18484
c@https://www.alphabounce.com/:3:14605

```

# Player Info
Allows script access: true

# Page Info
Page URL: https://www.alphabounce.com/

# Browser Info
Useragent: Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:84.0) Gecko/20100101 Firefox/84.0
OS: Win32

# Ruffle Info
Version: 0.1.0
Name: nightly 2021-01-11
Channel: nightly
Built: 2021-01-11T00:53:04.486Z
Commit: cce3a8199d48ecbce536a0e49efd501d97a2b370

**Comment by 9r8k** [comment]
Also, I have tried the two Nightly builds before that (i. e. 2021-01-10 and 2021-01-09). Still I get the same error page.

**Comment by 9r8k** [comment]
Another thought: Now I know that the game was created using Haxe. Possibly it runs via ActionScript 3 then, which I know Ruffle does not support yet.

**Comment by n0samu** [comment]
The panic is fixed, but the game still fails to load, showing a black screen with an "Invalid char at position 0" error:
![image](https://user-images.githubusercontent.com/71368227/192517314-3f8f2e35-ae7f-4a32-98c8-70819542f5fd.png)
This is the same problem that occurs when opening the [SWF URL](http://www.alphabounce.com/swf/game.swf?v=60) directly in either Ruffle or Adobe Flash Player. The SWF relies on flashvars as seen in the page source:
`so.addParam("FlashVars","swfUrl=/swf/&demo=1&lang=en");`
Without the flashvars, it requests its secondary asset `planet.swf` from the wrong URL:
Correct: `https://www.alphabounce.com/swf/planet.swf`
Incorrect: `http://www.alphabounce.com/web/www/swf/planet.swf`
When the flashvars are included in the SWF URL, the game works fine in Ruffle! `https://www.alphabounce.com/swf/game.swf?v=60&swfUrl=/swft/&demo=1&lang=en`
All of this made me think that Ruffle was not correctly passing the flashvars from the embed to the game. But when I reproduced the webpage in my local testing environment, the problem disappeared! I used the same HTML code and folder structure as the original site. The only explanation that I can think of is that some JavaScript on the site must be subtly interfering with Ruffle, preventing the flashvars from working. I don't know how this would even be possible.

**Comment by n0samu** [comment]
Original issue is fixed and the issue I pointed out is fixed by #8538. Closing
